vispy/visuals/image.py

```diff
@@ -451,8 +105,11 @@ def set_data(self, data, offset=None, copy=False):
     vec4 apply_clim(vec4 color) {
         if ($clim.x < $clim.y) {{
             color.rgb = clamp(color.rgb, $clim.x, $clim.y);
-        }} else {{
+        }} else if ($clim.x > $clim.y) {{
             color.rgb = clamp(color.rgb, $clim.y, $clim.x);
+        }} else {{
+            // clims are the same, show minimum colormap value
+            return vec4(0.0, 0.0, 0.0, 1.0);
```

Comment:
To avoid branching in the shader, it would be good to do checks like this on the CPU (in clim_normalized) and output a clim that can be used in the shader using a one-liner. (Branches can make shaders harder to run in parallel). But we may do that in another PR.

Reply:
> CPU (in clim_normalized) and output a clim that can be used in the shader using a one-liner.

Not sure what you mean with the one-liner comment. I agree and understand that branches should be avoided, but I'm not sure any clim on the CPU can be produced that makes the whole image 0. That means the data would have to be uploaded to be all 0s and I was trying to avoid having to reupload or change the texture data. Other option?

Reply:
Sorry for being a bit cryptic. I mean that the logic here in the shader can become a onelined without control flow (if's).

For the CPUScaledTexture we already scale the data, so we can e.g. subtract the (common) clim value, and set clim `(value, value+1)`. For the GPUScaledTexture maybe make the second clim value huge?

Reply:
Anyway, let's not make this hold up the current PR.

Reply:
I could have a look at that clim tweaking after this is merged, and also include Eric's solution of #1769, perhaps.

Reply:
That sounds good. I still feel weird about that whole discussion. I've been working with dask so long the idea of operations that require looking at the entire array at once feel like a bad idea. Anyway...we can resolve this comment then right? Anything left on this PR?

Reply:
I think it should be good to go? Github says that Rougier has requested changes, have you covered those?

Reply:
Yes. Quite a bit has actually changed since his review and I haven't heard anything lately. Given my urgency to make a release and include this news in the CZI report I have due at the end of the month, I'd like to merge this today or tomorrow. @rougier if you have time to re-review it that'd be great, otherwise I'll plan on merging it tomorrow.

Reply:
Correction: turns out my report isn't do until September 1st. Even though I have more time I'd still like to release 0.7 next week which means merging this this week.